### PR TITLE
Fix provisioners with Fingerbank

### LINKED ISSUE #503: Provisioners OS detection with Fingerbank
Since Fingerbank integration, we are no longer detecting OS type correctly for provisioners purposes.

### PR DESCRIPTION
**Pull request by dwlfrth**
## Description

Fixing 'provisioners' with Fingerbank
- Need fingerbank-1.0.1

Please see Fingerbank commit
https://github.com/inverse-inc/fingerbank/commit/6291fa306528d2aa68168d3ac2eb2857c236dc4f
## Impacts

Provisioners flow
## Delete branch after merge

YES 
## Issue

fixes #503
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
- Provisioners OS detection with Fingerbank (#503)

### CODE REVIEW

**Comment by julsemaan** [comment]
@lzammit said it works

File: lib/pf/provisioner.pm [hunk 1]
```diff
@@ -157,8 +158,12 @@ sub matchCategory {
...
-    my ($self, $os) = @_;
+    my ($self, $device_type) = @_;
     my @oses = @{$self->oses || []};
+
+    # Get device type kind of device by querying Fingerbank
+    my $os = pf::fingerbank::is_a($device_type);
+
     #if if no oses are defined then it will match all the oses
     local $_;
     return @oses == 0 || any { $os =~ $_ } @oses;
```
Comment (julsemaan) [inline lib/pf/provisioner.pm]: Shouldn't it become '$os eq $_' since the OS is not a part of the string anymore

**Comment by jrouzierinverse** [comment]
Can we have a test?

**Comment by dwlfrth** [comment]
@jrouzierinverse said he will add a test for Android

**Comment by dwlfrth** [comment]
Test added.
Please re-review

**Comment by dwlfrth** [comment]
Do not merge yet since new Fingerbank packages are needed.

**Comment by dwlfrth** [comment]
Please review

File: html/pfappserver/lib/pfappserver/Form/Config/Provisioning.pm [hunk 2]
```diff
@@ -118,9 +118,9 @@ sub ACCEPT_CONTEXT {
     my ($self, $c, @args) = @_;
     my ($status, $roles) = $c->model('Roles')->list();
     my @oses = ["Windows" => "Windows",
-                "Mac OS" => "Mac OS",
-                "Android" => "Android", 
-                "Apple" => "Apple IOS device"
+                "Macintosh" => "Mac OS",
+                "Generic Android" => "Android", 
+                "Apple iPod, iPhone or iPad" => "Apple IOS device"
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Provisioning.pm]: Also replace _IOS_ by _iOS_.

File: html/pfappserver/lib/pfappserver/Form/Config/Provisioning.pm [hunk 3]
```diff
@@ -118,9 +118,9 @@ sub ACCEPT_CONTEXT {
     my ($self, $c, @args) = @_;
     my ($status, $roles) = $c->model('Roles')->list();
     my @oses = ["Windows" => "Windows",
-                "Mac OS" => "Mac OS",
-                "Android" => "Android", 
-                "Apple" => "Apple IOS device"
+                "Macintosh" => "Mac OS",
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Provisioning.pm]: I would replace _Mac OS_ by _Mac OS X_.

**Comment by lzammit** [comment]
Tested with provionner for iOS and it worked.

Looks good to me.

**Comment by extrafu** [comment]
Does it also work after a reboot?

**Comment by julsemaan** [comment]
With @lzammit everything works with a reboot.

**Comment by dwlfrth** [comment]
As we discussed, the reviewer should change the label to 'reviewed - ready to merge' and merge.

**Comment by louismunro** [comment]
Wait before merging this one though.  
We need to fix the broken devel build before merging.

## 

LM

**Pull request merged**